documentation/specs/FocusedProductLayout.md

```diff
@@ -0,0 +1,215 @@
+
+`FocusedProductLayout` can render either an empty content section using `FocusedProductLayout.Content`, or a wizard content section using `FocusedProductLayout.WizardContent`.
+
+`FocusedProductLayout.WizardContent` utilizes the `Stepper` component to provide accessible step visualization across a defined number of wizard steps. It also includes buttons below the primary content area for navigating between steps.
+
+`FocusedProductLayout` accepts a `renderSidePanel` render prop to optionally render an inverse background side panel within the main content area. The side panel can be positioned to be at the `start` or `end` of the main content area. It defaults to the `end`.
+
+`FocusedProductLayout.Header` accepts a title, help menu items, and [render props](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop) for a back button and a logo. Utilizing render props allows for the rendering of each item to remain flexibile while passing in any unique design constraint. `renderBackArrow`, for instance, can support the back button being a true `button` or a framework-specific `Link` component.
+
+### API
+
+```ts
+export type FocusedProductLayoutProps = {
+  children: ReactNode;
+  renderSidePanel?: () => ReactNode;
+  sidePanelPosition?: "start" | "end";
+};
+
+export type FocusedProductLayoutContentProps = {
+  children: ReactNode;
+};
+
+export type FocusedProductLayoutWizardContentProps = {
+  children: ReactNode;
+  nextAction: FocusedProductLayoutWizardContentActionProps;
+  previousAction: FocusedProductLayoutWizardContentActionProps;
+  stepper: ReactNode;
+};
+
+type FocusedProductLayoutWizardContentActionProps = {
+  content: string;
+  onAction: () => void;
+  isDisabled?: boolean;
+};
+
+export type FocusedProductLayoutHeaderProps = {
+  helpMenuItems: ReactNode[];
+  renderBackArrow: (props: { children: ReactNode }) => ReactNode;
+  renderLogo: () => ReactNode;
+  title: ReactNode;
+};
+```
+
+### Example Usage
+
+_Simple content:_
+
+```tsx
+import { FocusedProductLayout } from "@easypost/easy-ui/ProductLayout";
+
+function App() {
+  return (
+    <FocusedProductLayout>
+      <FocusedProductLayout.Header
+        helpMenuItems={[
+          <Menu.Item key="1" href="https://easypost.com" target="_blank">
+            Documentation
+          </Menu.Item>,
+          // etc
+        ]}
+        renderBackArrow={(props) => {
+          // pass props that represent any unique design constraints
+          // in this case, they are `BackArrowIcon` as `children`
+          return (
+            <Link href="/account/settings" passHref>
+              <a>{props.children}</a>
+            </Link>
+          );
+        }}
+        renderLogo={() => <EasyPostLogo />}
```

Comment:
i went back and forth with these props as render props.

<img width="343" alt="image" src="https://github.com/EasyPost/easy-ui/assets/752942/81ff6070-cf6f-486a-a152-1ba7c6614169">

for `renderLogo`, since our library is open source, i figured it would be best to not embed our logo into the component. though the sizing constraints for the logo will be built into the logo container in this component.

for `renderBackArrow`, while most of the time this will be a client-side navigation using `Link` or another framework-specific navigation component, it could foreseeably be a button that simply changes application state, so i figured leaving flexibility to the consumer would be good. alternatively we could have specific props to enforce it being a link and/or other specific behavior, such as `backArrowHref` and `backArrowComponent` but that feels a little clunk/rigid to me.